tachyon/crypto/commitments/fri/two_adic_fri_prover.h

```diff
@@ -115,14 +115,12 @@ std::vector<CommitPhaseProofStep<MMCS>> AnswerQuery(
 }
 
 template <typename InputMMCS, typename ExtF, typename ChallengeMMCS,
-          typename Challenger,
-          typename Function = base::RepeatingCallback<
-              std::vector<BatchOpening<InputMMCS>>(size_t)>,
+          typename Challenger, typename OpenInput,
           typename F = typename math::ExtensionFieldTraits<ExtF>::BaseField>
 TwoAdicFriProof<ChallengeMMCS, std::vector<BatchOpening<InputMMCS>>, F>
 TwoAdicFriPcsProve(TwoAdicFriConfig<ChallengeMMCS>& config,
                    std::vector<std::vector<ExtF>>&& inputs,
-                   Challenger& challenger, Function open_input) {
+                   Challenger& challenger, OpenInput open_input) {
```

Comment:
I think it would be better if the name clearly indicated that it is a function pointer. How about something like 'OpenInputCallback'?